Boundries Assignment/CoordinatesFinder/Program.cs

```
@@ -0,0 +1,61 @@
+﻿using CoordinatesFinder.Constants;
+using CoordinatesFinder.Exceptions;
+using CoordinatesFinder.Helpers;
+using CoordinatesFinder.Models;
+using Microsoft.Extensions.Configuration;
+
+
+namespace CoordinatesFinder
+{
+    public class CoordinatesFinder
+    {
+        public static async Task Main(string[] args)
+        {
+            IConfiguration configuration = new ConfigurationBuilder()
+                .SetBasePath(Directory.GetCurrentDirectory()).AddJsonFile(AppConstants.AppSettingsJson).Build();
+
+            HttpClient client = new HttpClient();
+            CoordinatesFinderHelper coordinatesFinder = new CoordinatesFinderHelper(client, configuration);
+
+            try
+            {
+                var location = GetLocation();
+                Coordinates coordinates = await coordinatesFinder.GetCoordinatesFromLocationAsync(location);
+                Console.WriteLine($"Latitude: {coordinates.Latitude}, Longitude: {coordinates.Longitude}");
+            }
+            catch (GeocodingException ex)
+            {
+                Console.WriteLine($"Error: {ex.Message}");
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"Error: {ex.Message}");
+            }
```

I think you do not need two catch block here, the second catch (Exception ex) block will catch all exceptions, including GeocodingException, unless it is already caught by the previous block.

---

Yes, For Current Situation you are correct here, since GeocodingException doesn't do much apart from message part. For this situation, I can say Exception will work.